--- origsrc/SDL-1.2.14/configure.in	2009-10-17 12:35:03.000000000 -0500
+++ src/SDL-1.2.14/configure.in	2009-11-17 00:56:58.602877400 -0600
@@ -77,11 +77,6 @@ if test x$srcdir != x.; then
     INCLUDE="-Iinclude $INCLUDE"
 fi
 case "$host" in
-    *-*-cygwin*)
-        # We build SDL on cygwin without the UNIX emulation layer
-        BASE_CFLAGS="-I/usr/include/mingw -mno-cygwin"
-        BASE_LDFLAGS="-mno-cygwin"
-        ;;
     *)
         BASE_CFLAGS="-D_GNU_SOURCE=1"
         BASE_LDFLAGS=""
@@ -157,7 +152,7 @@ if test x$enable_libc = xyes; then
 
     dnl Checks for library functions.
     case "$host" in
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ;;
     *)
         AC_FUNC_ALLOCA
@@ -730,7 +725,7 @@ AC_HELP_STRING([--enable-nasm], [use nas
 
         if test x"$NASMFLAGS" = x; then
             case $ARCH in
-              win32)
+              win32|cygwin)
                   NASMFLAGS="-f win32"
                   ;;
               openbsd)
@@ -965,6 +960,13 @@ AC_HELP_STRING([--enable-x11-shared], [d
                           , enable_x11_shared=maybe)
 
             case "$host" in
+                *-*-cygwin*)
+                    x11_symbols_private=yes
+                    x11_lib='cygX11-6.dll'
+                    x11ext_lib='cygXext-6.dll'
+                    xrender_lib='cygXrender-1.dll'
+                    xrandr_lib='cygXrandr-2.dll'
+                    ;;
                 *-*-darwin*) # Latest Mac OS X actually ships with Xrandr/Xrender libs...
                     x11_symbols_private=yes
                     x11_lib='/usr/X11R6/lib/libX11.6.dylib'
@@ -2453,7 +2455,7 @@ case "$host" in
             have_timers=yes
         fi
         ;;
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ARCH=win32
         if test "$build" != "$host"; then # cross-compiling
             # Default cross-compile location
@@ -2525,6 +2527,44 @@ case "$host" in
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
         SDL_LIBS="-lmingw32 -lSDLmain $SDL_LIBS -mwindows"
         ;;
+    *-*-cygwin*)
+        ARCH=cygwin
+        CheckDummyVideo
+        CheckDiskAudio
+        CheckDummyAudio
+        CheckDLOPEN
+        CheckNASM
+        CheckOSS
+        CheckARTSC
+        CheckESD
+        CheckNAS
+        CheckX11
+        CheckGGI
+        CheckAAlib
+        CheckCaca
+        CheckOpenGLX11
+        CheckPTHREAD
+        # Set up files for the joystick library
+        if test x$enable_joystick = xyes; then
+            AC_DEFINE(SDL_JOYSTICK_WINMM)
+            SOURCES="$SOURCES $srcdir/src/joystick/win32/*.c"
+            have_joystick=yes
+        fi
+        # Set up files for the cdrom library
+        if test x$enable_cdrom = xyes; then
+            AC_DEFINE(SDL_CDROM_WIN32)
+            SOURCES="$SOURCES $srcdir/src/cdrom/win32/*.c"
+            have_cdrom=yes
+        fi
+        # Set up files for the timer library
+        if test x$enable_timers = xyes; then
+            AC_DEFINE(SDL_TIMER_UNIX)
+            SOURCES="$SOURCES $srcdir/src/timer/unix/*.c"
+            have_timers=yes
+        fi
+        # Set up the system libraries we need
+        EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lwinmm"
+        ;;
     *-wince*)
         ARCH=win32
         CheckDummyVideo
--- origsrc/SDL-1.2.14/include/SDL_platform.h	2009-10-12 18:07:19.000000000 -0500
+++ src/SDL-1.2.14/include/SDL_platform.h	2009-11-17 00:56:58.602877400 -0600
@@ -43,6 +43,10 @@
 #undef __BSDI__
 #define __BSDI__	1
 #endif
+#if defined(__CYGWIN__)
+#undef __CYGWIN__
+#define __CYGWIN__	1
+#endif
 #if defined(_arch_dreamcast)
 #undef __DREAMCAST__
 #define __DREAMCAST__	1
--- origsrc/SDL-1.2.14/src/video/x11/SDL_x11gl.c	2009-10-17 02:45:42.000000000 -0500
+++ src/SDL-1.2.14/src/video/x11/SDL_x11gl.c	2009-11-17 00:56:58.602877400 -0600
@@ -35,6 +35,8 @@
 #define DEFAULT_OPENGL	"libGL.so.3"
 #elif defined(__OpenBSD__)
 #define DEFAULT_OPENGL	"libGL.so.4.0"
+#elif defined(__CYGWIN__)
+#define DEFAULT_OPENGL	"cygGL-1.dll"
 #else
 #define DEFAULT_OPENGL	"libGL.so.1"
 #endif
